barter-data/src/streams/builder/dynamic/mod.rs

```diff
@@ -107,639 +139,461 @@ impl<InstrumentKey> DynamicStreams<InstrumentKey> {
+                                    subs
+                                )
+                            }
+                            (ExchangeId::GateioPerpetualsUsd, SubKind::PublicTrades) => {
+                                init_stream_logic!(
+                                    GateioPerpetualsUsd::default(),
+                                    PublicTrades,
+                                    txs,
+                                    exchange,
+                                    trades,
+                                    subs
+                                )
+                            }
+                            (ExchangeId::GateioPerpetualsBtc, SubKind::PublicTrades) => {
+                                init_stream_logic!(
+                                    GateioPerpetualsBtc::default(),
+                                    PublicTrades,
+                                    txs,
+                                    exchange,
+                                    trades,
+                                    subs
+                                )
+                            }
+                            (ExchangeId::GateioOptions, SubKind::PublicTrades) => {
+                                init_stream_logic!(
+                                    GateioOptions::default(),
+                                    PublicTrades,
+                                    txs,
+                                    exchange,
+                                    trades,
+                                    subs
+                                )
+                            }
+                            (ExchangeId::Kraken, SubKind::PublicTrades) => {
+                                init_stream_logic!(
+                                    Kraken,
+                                    PublicTrades,
+                                    txs,
+                                    exchange,
+                                    trades,
+                                    subs
+                                )
+                            }
+                            (ExchangeId::Kraken, SubKind::OrderBooksL1) => {
+                                init_stream_logic!(Kraken, OrderBooksL1, txs, exchange, l1s, subs)
+                            }
+
+                            // Fallback for unsupported combinations
+                            (exchange, sub_kind) => {
+                                // Use different names for clarity
+                                Err(DataError::Unsupported { exchange, sub_kind })
+                            }
+                        }
+                    }
+                });
+
+        try_join_all(batch_futures).await?;
+        Ok(BatchedStreams {
             trades: channels
                 .rxs
                 .trades
                 .into_iter()
-                .map(|(exchange, rx)| (exchange, rx.into_stream()))
+                .map(|(ex, rx)| (ex, rx.into_stream()))
```

Comment:
I would keep a longer binding name here. It's more descriptive.
```suggestion
                .map(|(exchange, rx)| (exchange, rx.into_stream()))
```